source/2_platform/rex_windows/src/gui_application.cpp

```diff
@@ -128,14 +127,21 @@ namespace rex
 
       void subscribe_window_events()
       {
-        event_system::subscribe(event_system::EventType::WindowClose, [this](const event_system::Event& /*evt*/) { m_app_instance->quit(); });
+        event_system::subscribe(event_system::EventType::WindowClose,
+                                [this](const event_system::Event& /*evt*/)
+                                {
+                                  rex::event_system::Event ev {};
+                                  ev.type = rex::event_system::EventType::QuitApp;
+                                  rex::event_system::fire_event(ev);
```

Comment:
This is a direct call IIRC?
You should be careful when calling events from another event.
The event_system should queue events and poll for available events ( maybe even a max per frame ) 
No part of this PR but we should create an issue for it.

Reply:
I think events should be processed on a different tick than when they're added
events added on tick n should be processed on tick n + 1 to avoid any issues
I don't thin we should limit the number of events we can process in a frame though, maybe just warn if it's too many

Reply:
Well, that's what I'm saying right queue them up.